### DISCUSSION TITLE
Add CLI option to create / update database using a connection string from arguments

### DISCUSSION THREAD

**Discussion by Xzelsius** [JasperFx/marten#2746] [Ideas]
Based on [Command Line Tooling](https://martendb.io/configuration/cli.html#command-line-tooling)

Would it be possible to have a new argument for the `marten-apply` / `marten-assert` / etc to pass in a connection string?

In our case, we use dynamic tenant databases (meaning we only have the master connection string configured).
So we can't use `Apply...OnStartup()` nor can we use the CLI to migrate our databases.
The only option we have is database migration on the first request to said database, which is not ideal.

It would be great if we could pass a connection string to `marten-apply` so we can migrate each tenant database individually.

**Comment by jeremydmiller** [comment]
@Xzelsius You can already choose a single database. Another option that would work today would be to pass the connection string as an IConfiguration value, because you can override individual configuration values from the Oakton command line args.

I'm converting this to a discussion for now.

**Reply by Xzelsius** [reply]
Does your suggestion work with dynamic databases where we have only configured the master connection string?
Or would we have to build a "switch" that decides between single tenant database and multi-tenancy based on the parameter?
